### PR TITLE
Updated with a bunch of hooks

### PR DESCRIPTION
**Pull request by PsychoTea**
Loads of new hooks (mainly in the player hooks category) including
OnPlayerHealthChange, OnPlayerSpectate(End), OnPlayerTick,
OnPlayerWounded

### CODE REVIEW

File: source/includes/rust/hooks_player.md [hunk 1]
```diff
@@ -233,6 +306,77 @@ void OnPlayerRespawned(BasePlayer player)
  * This means it's possible for a player to connect and disconnect from a server without OnPlayerRespawned ever triggering for them
  * No return behavior
 
+## OnPlayerSleep
+
+``` csharp
+void OnPlayerSleep(BasePlayer player)
+{
+	Puts("OnPlayerSleep works!");
```
Comment (lukespragg) [inline source/includes/rust/hooks_player.md]: Spaces, not tabs. ;)

File: source/includes/rust/hooks_player.md [hunk 2]
```diff
@@ -198,6 +235,18 @@ void OnPlayerInput(BasePlayer player, InputState input)
 ## OnPlayerLanded
 
 ``` csharp
+void OnPlayerLand(BasePlayer player)
+{
+	Puts("OnPlayerLand works!");
```
Comment (lukespragg) [inline source/includes/rust/hooks_player.md]: Spaces, not tabs.

File: source/includes/rust/hooks_player.md [hunk 3]
```diff
@@ -133,7 +145,7 @@ void OnPlayerAttack(BasePlayer attacker, HitInfo info)
  * Useful for modifying an attack before it goes out
  * hitInfo.HitEntity should be the entity that this attack would hit
  * Returning true cancels the attack
-
+
```
Comment (lukespragg) [inline source/includes/rust/hooks_player.md]: Unnecessary space.

File: source/includes/rust/hooks_item.md [hunk 4]
```diff
@@ -337,4 +337,4 @@ void OnTrapTrigger(BaseTrap trap, GameObject go)
 ```
 
  * Called when a trap is triggered by a game object
- * No return behavior
+ * No return behavior
```
Comment (lukespragg) [inline source/includes/rust/hooks_item.md]: What was the actual change here?

File: source/includes/rust/hooks_player.md [hunk 5]
```diff
@@ -233,6 +306,77 @@ void OnPlayerRespawned(BasePlayer player)
...
+ * Called when the player awakes
+ * No return behavior
+ 
+## OnPlayerSpectate
+
+``` csharp
+void OnPlayerSpectate(BasePlayer player, string spectateFilter)
+{
+	Puts("OnPlayerSpectate works!");
```
Comment (lukespragg) [inline source/includes/rust/hooks_player.md]: Spaces, not tabs.

File: source/includes/rust/hooks_player.md [hunk 6]
```diff
@@ -233,6 +306,77 @@ void OnPlayerRespawned(BasePlayer player)
...
+
+ * Called when a player starts spectating
+ * Returning a non-null value cancels the spectate
+ 
+## OnPlayerSpectateEnd
+
+``` csharp
+void OnPlayerSpectateEnd(BasePlayer player, string spectateFilter)
+{
+	Puts("OnPlayerSpectateEnd works!");
```
Comment (lukespragg) [inline source/includes/rust/hooks_player.md]: Spaces, not tabs.

File: source/includes/rust/hooks_item.md (hunk 4 above)
Comment (PsychoTea) [inline source/includes/rust/hooks_item.md]: I have no idea. Not sure why GitHub picked up a change. I didn't touch the traps etc.

File: source/includes/rust/hooks_player.md [hunk 7]
```diff
@@ -233,6 +306,77 @@ void OnPlayerRespawned(BasePlayer player)
...
+ 
+ * Called when a player stops spectating
+ * Returning a non-null value stops the spectate from ending
+
+## OnPlayerTick
+
+``` csharp
+void OnPlayerTick(BasePlayer player, PlayerTick msg, bool wasPlayerStalled)
+{
+	Puts("OnPlayerTick works!");
```
Comment (lukespragg) [inline source/includes/rust/hooks_player.md]: Spaces, not tabs.

File: source/includes/rust/hooks_player.md [hunk 8]
```diff
@@ -233,6 +306,77 @@ void OnPlayerRespawned(BasePlayer player)
...
+```
+
+ * Returning a non-null value overrides default behavior
+ 
+## OnPlayerWound
+
+``` csharp
+void OnPlayerWound(BasePlayer player)
+{
+	Puts("OnPlayerWound works!");
```
Comment (lukespragg) [inline source/includes/rust/hooks_player.md]: Spaces, not tabs.

**Comment by PsychoTea** [comment]
Fixed everything :)

File: source/includes/rust/hooks_item.md (hunk 4 above)
Comment (Mughisi) [inline source/includes/rust/hooks_item.md]: There was a space after `behavior` by the looks of it

**Pull request merged**